extensions/ql-vscode/src/query-history/store/data-mapper.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  LocalQueryInfo,
+  CompletedQueryInfo,
+  InitialQueryInfo,
+} from "../../query-results";
+import { QueryEvaluationInfo } from "../../run-queries-shared";
+import { QueryHistoryInfo } from "../query-history-info";
+import { VariantAnalysisHistoryItem } from "../variant-analysis-history-item";
+import {
+  CompletedQueryInfoData,
+  QueryEvaluationInfoData,
+  InitialQueryInfoData,
+  LocalQueryDataItem,
+} from "./local-query-data-item";
+import { QueryHistoryDataItem } from "./query-history-data";
+
+// Maps Query History Data Models to Domain Models
+
+export function mapQueryHistoryToDomainModels(
+  queries: QueryHistoryDataItem[],
+): QueryHistoryInfo[] {
+  return queries.map((d) => {
+    if (d.t === "variant-analysis") {
+      const query: VariantAnalysisHistoryItem = d;
+      return query;
+    } else if (d.t === "local") {
+      return mapLocalQueryDataItemToDomainModel(d);
+    }
+
+    throw Error(
+      `Unexpected or corrupted query history file. Unknown query type: ${d}`,
```

Comment:
```suggestion
      `Unexpected or corrupted query history file. Unknown query history item: ${JSON.stringify(d)}`,
```
Thanks for updating, though I'm afraid this will print `[Object object]`. We could stringify the item like this. It could be a lot of text, but it would at least include all the info needed to debug it.

Reply:
👍 comitted your suggestion